src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -81,6 +81,26 @@ public static BooleanExpression isFilePresentForSelectedEntry(StateManager state
         return BooleanExpression.booleanExpression(fileIsPresent);
     }
 
+    /**
+     * Check if at least one of the selected entries has linked files
+     * <br>
+     * Used in {@link org.jabref.gui.maintable.OpenExternalFileAction} when multiple entries selected
+     * @param stateManager manager for the state of the GUI
+     * @param preferencesService services of preferences
+     * @return a boolean binding
+     */
+    public static BooleanExpression hasLinkedFileForSelectedEntries(StateManager stateManager, PreferencesService preferencesService) {
+        return Bindings.createBooleanBinding(() -> {
```

Comment:
This will result in a constant binding, i.e. changing the selected entries will not update the binding. For this to work, you need to specify `selectedEntries` also as a second parameter. Or even better use `EasyBind.reduce(statemanager.getSelectedEntries(), entries -> entries.anyMatch(entry -> !entry.getFiles().isEmpty())`

Reply:
Frankly I did not see any wrong behavior of the binding. See the video `6966-4` above: first I selected three entries with no linked file, the option is disabled. And then I selected another entry which has, then the option is enabled. Maybe I misunderstood your explanation.

Anyway, I modified the boolean binding as you suggested. Now the code is much conciser, and the behavior remains normal. Besides, I'm not so sure about the usage of `EasyBind`, are my codes correct?

Thanks again for your advice!

Reply:
To be honest, I also didn't understood why it worked before. Maybe the binding was re-created every time? Not sure.

Anyway, the new code looks good. Thanks for the update.